core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,224 @@
+      function() {
+        return topicInitializedEventEmitter;
+      });
+    spyOnProperty(
+      TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+      function() {
+        return topicReinitializedEventEmitter;
+      });
+    ctrl.$onInit();
+  }));
+
+  afterEach(() => {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize the variables', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    spyOn(TopicEditorStateService, 'getSkillIdToRubricsObject').and
+      .returnValue(skillIdToRubricsObject);
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+    expect($scope.selectedSkillId).toBeNull();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.skillIdToRubricsObject).toEqual(skillIdToRubricsObject);
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.getSkillsCategorizedByTopics).toBe(CategorizedSkillsDict);
+    expect($scope.getUntriagedSkillSummaries)
+      .toBe(UntriagedSkillSummaries);
+    expect($scope.canEditQuestion).toBe(false);
+    expect($scope.misconceptions).toEqual([]);
+    expect($scope.questionIsBeingUpdated).toBe(false);
+    expect($scope.questionIsBeingSaved).toBe(false);
+    expect($scope.emptyMisconceptionsList).toEqual([]);
+  });
+
+  it('should setFocus on selectSkillField when screen loads', function() {
+    $window.onload();
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+  });
+
+  it('should call initTab when topic is initialized', function() {
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+
+    topicInitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should call initTab when topic is reinitialized', function() {
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+
+    topicInitializedEventEmitter.emit();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+    topicReinitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should unsubscribe when onDestroy runs', function() {
+    spyOn(ctrl.directiveSubscriptions, 'unsubscribe');
+
+    ctrl.$onDestroy();
+
+    expect(ctrl.directiveSubscriptions.unsubscribe).toHaveBeenCalled();
+  });
+
+  it('should reinitialize questions list', function() {
```

Comment:
This isn't a behaviour. What outcome are you expecting when the questions list is reinitialized? Test that.

Reply:
I'm testing this function here:
https://github.com/oppia/oppia/blob/ddfab3665410c41e72d16256d77862e9373b5377/core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.ts#L75-L81

The outcomes I'm expecting are the functions `QuestionsListService.resetPageNumber()` 
 and `QuestionsListService.getQuestionSummariesAsync` being called and the `$scope.selectedSkillId` being set.